rust/xain-proto/generate.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+cd "$(dirname "$0")"
+
+if ! [[ -x "$(command -v grpc_rust_plugin)" ]]; then
+    echo "Error: grpc_rust_plugin was not found"
+    echo
+    echo "To install, run: cargo install grpcio-compiler"
+    exit 1
+fi
+
+proto_files="
+../../numproto/numproto/protobuf/ndarray.proto
+../../protobuf/xain/grpc/coordinator.proto
+../../protobuf/xain/grpc/hellonumproto.proto
+"
+
+for proto in $proto_files; do
+    echo "Processing: $proto"
+    protoc \
+        --rust_out=$PWD/src \
+        --grpc_out=$PWD/src \
+        --plugin=protoc-gen-grpc=`which grpc_rust_plugin` \
+        --proto_path=../../numproto \
+        --proto_path=../../protobuf/xain/grpc \
+        $proto
+done
```

Comment:
Can we run this script on CI and check that the generated file is fresh?

Reply:
Not sure, maybe. A roadblock with that approach might be the fact that generated Rust files are not always perfect and need some small manual tweaks before they compile. 

We didn't run into any of those issues here, but I did last time I played with grpcio. One of the issues was that methods named `mod` were not properly escaped with `r#mod`.